.github/workflows/lint-and-test.yaml

```diff
@@ -28,16 +40,171 @@ concurrency:
+
+      - name: Set environment variables
+        run: |
+          echo "VSOMEIP_SOURCE_TARBALL=${{ env.VSOMEIP_RELEASE_URL }}${{ env.VSOMEIP_RELEASE_TAG }}" >> $GITHUB_ENV
+          echo "VSOMEIP_SOURCE_PATH=${{ github.workspace }}/${{ env.VSOMEIP_SOURCE_PATH }}" >> $GITHUB_ENV
+          echo "VSOMEIP_INSTALL_PATH=${{ github.workspace }}/${{ env.VSOMEIP_INSTALL_PATH }}" >> $GITHUB_ENV
+          echo "VSOMEIP_SOURCE_PATH_WITH_WILDCARD=${{ github.workspace }}/${{ env.VSOMEIP_SOURCE_PATH }}/*" >> $GITHUB_ENV
+
+      - name: Print environment variables for debugging
+        run: |
+          echo "VSOMEIP_SOURCE_TARBALL: ${{ env.VSOMEIP_SOURCE_TARBALL }}"
+          echo "VSOMEIP_SOURCE_PATH: ${{ env.VSOMEIP_SOURCE_PATH }}"
+          echo "VSOMEIP_INSTALL_PATH: ${{ env.VSOMEIP_INSTALL_PATH }}"
+          echo "VSOMEIP_SOURCE_PATH_WITH_WILDCARD: ${{ env.VSOMEIP_SOURCE_PATH_WITH_WILDCARD }}"
+
+      - name: Download tarball
+        run: |
+          wget -O vsomeip-source.tar.gz $VSOMEIP_SOURCE_TARBALL
+          ls -l
+
+      - name: Create destination directory
+        run: |
+          mkdir -p $VSOMEIP_SOURCE_PATH
+          ls -l
+
+      - name: Decompress tarball
+        run: |
+          tar --strip-components=1 -xzf vsomeip-source.tar.gz -C $VSOMEIP_SOURCE_PATH
+          ls -l
+
+      - name: Confirm decompressed
+        run: |
+          echo "Current working directory: $(pwd)"
+          echo "Contents of repository root:"
+          ls -l
+          echo "Contents of decompressed folder ($VSOMEIP_SOURCE_PATH):"
+          ls -l $VSOMEIP_SOURCE_PATH
+
+      - name: Generate cache key
+        run: echo "CACHE_KEY=${{ runner.os }}-vsomeip-${{ hashFiles(env.VSOMEIP_SOURCE_PATH_WITH_WILDCARD) }}" >> $GITHUB_ENV
+
+      - name: Set cache key output
+        id: generate_cache_key
+        run: echo "CACHE_KEY=${{ env.CACHE_KEY }}" >> $GITHUB_OUTPUT
+
+      - name: Cache vsomeip build
+        id: cache-vsomeip
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.VSOMEIP_INSTALL_PATH }}
+          key: ${{ runner.os }}-vsomeip-${{ hashFiles(env.VSOMEIP_SOURCE_PATH_WITH_WILDCARD) }}
+          restore-keys: |
+            ${{ runner.os }}-vsomeip-
+
+      - name: Install dependencies
+        if: steps.cache-vsomeip.outputs.cache-hit != 'true'
+        run: sudo apt-get install -y build-essential cmake libboost-all-dev
+
+      - name: Build vsomeip
+        if: steps.cache-vsomeip.outputs.cache-hit != 'true'
+        run: |
+          cd $VSOMEIP_SOURCE_PATH
+          cmake -Bbuild -DCMAKE_INSTALL_PREFIX=$VSOMEIP_INSTALL_PATH -DENABLE_SIGNAL_HANDLING=1 .
+          cmake --build build --target install
+
+      - name: Save vsomeip build to cache
+        if: steps.cache-vsomeip.outputs.cache-hit != 'true'
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.VSOMEIP_INSTALL_PATH }}
+          key: ${{ runner.os }}-vsomeip-${{ hashFiles(env.VSOMEIP_SOURCE_PATH_WITH_WILDCARD) }}
+
+  foo:
```

Comment:
maybe a more descriptive job name?

Reply:
Hah, sorry. I'm still in the midst of updating the CI. Feel free to review everything but that :)

Reply:
fair enough :)

Reply:
foo removed :)